cicecore/cicedynB/dynamics/ice_transport_driver.F90

```diff
@@ -689,6 +689,8 @@ subroutine transport_remap (dt)
                                         istop,              jstop)
 
                if (ckflag) then
+                  write (nu_diag,*) 'istep1, my_task, iblk, cat =',     &
+                                     istep1, my_task, iblk, n
```

Comment:
OK, here we will end up with duplicated printing of `istep1, my_task, iblk`, but we do not loose the category info. That's OK.
Out of curiosity, why not add the category as an argument to `diagnostic_abort` and `print_state` ? I'm guessing the answer is "to not have to update all callers of these subroutine", which I think does make sense.